docs/reference/aggregations/bucket/random-sampler-aggregation.asciidoc

```diff
@@ -0,0 +1,96 @@
+[[search-aggregations-random-sampler-aggregation]]
+=== Random sampler aggregation
+++++
+<titleabbrev>Random sampler</titleabbrev>
+++++
+
+experimental::[]
+
+The `random_sampler` aggregation is a single bucket aggregation that randomly
+includes documents in the aggregated results. Sampling provides significant
+speed improvement at the cost of accuracy.
+
+The sampling is accomplished by providing a random subset of the entire set of
+documents in a shard. If a filter query is provided in the search request, that
+filter is applied over the sampled subset. Consequently, if a filter is
+restrictive, very few documents might match; therefore, the statistics might not
+be as accurate.
+
+NOTE: This aggregation is not to be confused with the
+<<search-aggregations-bucket-sampler-aggregation,sampler aggregation>>. The
+sampler aggregation is not over all documents; rather, it samples the first `n`
+documents matched by the query.
+
+[source,console]
+----
+GET kibana_sample_data_ecommerce/_search?size=0&track_total_hits=false
+{
+  "aggregations": {
+    "sampling": {
+      "random_sampler": {
+        "probability": 0.1
+      },
+      "aggs": {
+        "price_percentiles": {
+          "percentiles": {
+            "field": "taxful_total_price"
+          }
+        }
+      }
+    }
+  }
+}
+----
+// TEST[setup:kibana_sample_data_ecommerce]
+
+[[random-sampler-top-level-params]]
+==== Top-level parameters for random_sampler
+
+`probability`::
+(Required, float) The probability that a document will be included in the
+aggregated data. Must be greater than 0, less than `0.5`, or exactly `1`. The
+lower the probability, the fewer documents are matched.
+
+`seed`::
+(Optional, integer) The seed to generate the random sampling of documents. When
+a seed is provided, the random subset of documents is the same between calls.
+
+[[random-sampler-inner-workings]]
+==== How does the sampling work?
+
+The aggregation is a random sample of all the documents in the index. In other
+words, the sampling is over the background set of documents. If a query is
+provided, a document is returned if it is matched by the query and if the
+document is in the random sampling. The sampling is not done over the matched
+documents.
+
+Consider the set of documents `[1, 2, 3, 4, 5]`. Your query matches `[1, 3, 5]`
+and the randomly sampled set is `[2, 4, 5]`. In this case, the document returned
+would be `[5]`.
+
+This type of sampling provides almost linear performance in relation to sampling
+probability.
```

Comment:
I know you qualify this below, but I think I would reword this slightly so it is a bit more explicit about the performance improvement. Also strictly linear in probability would imply proportional to p not 1 / p, so I would instead reference the reduction in sample document size.
```suggestion
This type of sampling provides almost linear improvement in query latency in relation to the amount
by which sampling reduces the document set size.
```